Terminal.Gui/Views/Button.cs

```diff
@@ -215,14 +233,36 @@ public override bool ProcessKey (KeyEvent kb)
 				return false;
 			}
 
-			var c = kb.KeyValue;
-			if (c == '\n' || c == ' ' || kb.Key == HotKey) {
-				Clicked?.Invoke ();
+			if (InvokeKeybindings (kb))
 				return true;
-			}
+
 			return base.ProcessKey (kb);
 		}
 
+		bool ExecuteHotKey (KeyEvent ke)
+		{
+			if (ke.IsAlt) {
```

Comment:
@BDisp why do we need this here when the keybinding for it is:

```csharp
AddKeyBinding (Key.AltMask | HotKey, Command.ExecuteHotKey);
```

I think this is the only place we currently check the KeyEvent of the `Func<KeyEvent, bool>`.  If we can eliminate it then we can make it `Func<bool>` which I think will be more maintainable and useful later on?

Reply:
Agree, go forward. Thanks.